demos/common/hfsm/az_iot_hfsm.c

```diff
@@ -0,0 +1,324 @@
+ */
+
+#include <stdint.h>
+#include <inttypes.h>
+
+#include "az_hfsm.h"
+#include "az_hfsm_pal_timer.h"
+#include "az_iot_hfsm.h"
+
+// TODO: change logging to az_log_internal.
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "IoT_HFSM"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+/* 
+ * The function prints to the console before the network is connected;
+ * then a UDP port after the network has connected. */
+extern void vLoggingPrintf( const char * pcFormatString,
+                            ... );
+
+/* Map the SdkLog macro to the logging function to enable logging */
+#ifndef SdkLog
+    #define SdkLog( message )    vLoggingPrintf message
+#endif
+
+#include "logging_stack.h"
+/**************************************************/
+
+const az_hfsm_event az_hfsm_event_az_iot_start = { AZ_IOT_START, NULL };
+#ifdef AZ_IOT_HFSM_PROVISIONING_ENABLED
+const az_hfsm_event az_hfsm_event_az_iot_provisioning_done = { AZ_IOT_PROVISIONING_DONE, NULL };
+#endif
+
+static int32_t azure_iot(az_hfsm* me, az_hfsm_event event);
+static int32_t idle(az_hfsm* me, az_hfsm_event event);
+#ifdef AZ_IOT_HFSM_PROVISIONING_ENABLED
+static int32_t provisioning(az_hfsm* me, az_hfsm_event event);
+#endif
+static int32_t hub(az_hfsm* me, az_hfsm_event event);
+
+// Hardcoded AzureIoT hierarchy structure
+static az_hfsm_state_handler azure_iot_hfsm_get_parent(az_hfsm_state_handler child_state)
+{
+  az_hfsm_state_handler parent_state;
+
+  if ((child_state == azure_iot))
+  {
+    parent_state = NULL;
+  }
+  else if (
+#ifdef AZ_IOT_HFSM_PROVISIONING_ENABLED    
+      (child_state == provisioning) ||
+#endif
+      (child_state == hub) || (child_state == idle)
+    )
+  {
+    parent_state = azure_iot;
+  }
+  else {
+    // Unknown state.
+    az_iot_hfsm_pal_critical_error(NULL);
+    parent_state = NULL;
+  }
+
+  return parent_state;
+}
+
+// AzureIoT
+static int32_t azure_iot(az_hfsm* me, az_hfsm_event event)
+{
+  int32_t ret = 0;
+  int32_t operation_msec;
+
+  az_iot_hfsm_type* this_iothfsm = (az_iot_hfsm_type*)me;
+
+  switch ((int32_t)event.type)
+  {
+    case AZ_HFSM_ENTRY:
+      LogInfo( ("AzureIoT: Entry") );
+      this_iothfsm->_use_secondary_credentials = false;
+      break;
+
+    case AZ_IOT_ERROR:
+      LogInfo( ("AzureIoT: AZ_IOT_ERROR") );
```

Comment:
LogInfo( ("AzureIoT: IoT_Error!") );

Reply:
Changing to standardized az_log functions and text.